pontoon/base/static/js/locale_project.js

```diff
@@ -0,0 +1,65 @@
+$(function() {
+
+  // Search entire project with Enter (and remove default search handler)
+  $('.part .menu .search-project input[type=search]').off('keyup.search').keydown(function (e) {
+    if (e.which === 13) {
+      e.stopPropagation();
+
+      var q = $(this).val(),
+          action = '';
+
+      if (q.length > 0) {
+        if ($('.locale .selector .language').length) {
+          var locale = $('.locale .selector .language').attr('class').split(' ')[1],
+              project = $('.project .selector .title').data('slug'),
+              action = '/' + locale + '/' + project + '/';
+        }
+
+        var options = '';
+        $('.search-project .options li').each(function() {
+          if ($(this).is('.enabled')) {
+            options += '&' + $(this).data('name') + '=' + $(this).data('name');
+          }
+        });
+
+        // At least one option must be selected
+        if (!options) {
+          return false;
+        }
+
+        var post = $('<form>', {
+          method: 'post',
+          action: action + '?q=' + q + options
+        });
+
+        $('<input>', {
+          type: 'hidden',
+          name: 'csrfmiddlewaretoken',
+          value: $('#server').data('csrf')
+        }).appendTo(post);
+
+        post.appendTo('body').submit().remove();
+        return false;
+      }
+    }
+  });
+
+  // Focus search field when switching tabs
+  $(".tabs nav a").click(function (e) {
+    $('.part .menu input[type=search]:visible').trigger("keyup").focus();
+  });
+
+  // If showing search results, show search pane by default
+  var search = $('#server').data("search"),
+      tab = 'resources';
+  if (search) {
+    tab = 'search-project';
+  }
+  $('.part .menu nav a[href=#' + tab + ']').click();
```

Comment:
It's fine for this PR because I still think the frontend needs a major refactoring rather than incremental change, but one of the anti-patterns I've seen a lot in Pontoon is simulating events like this to trigger other functions. What you should really be doing is putting the logic that you want to reuse in a function and having both handlers call that function instead. Simulating events like this makes it very hard to figure out what piece of code is being executed when this line is called.